client/web/src/enterprise/batches/create/ConfigurationForm.tsx

```diff
@@ -0,0 +1,211 @@
+        setIsNameValid(NAME_PATTERN.test(event.target.value))
+    }, [])
+
+    const history = useHistory()
+    const handleCancel = (): void => history.goBack()
+    const handleCreate = (): void => {
+        const redirectSearchParameters = new URLSearchParams()
+        if (insightTitle) {
+            redirectSearchParameters.set('title', insightTitle)
+        }
+        let serializedRedirectSearchParameters = redirectSearchParameters.toString()
+        if (serializedRedirectSearchParameters.length > 0) {
+            serializedRedirectSearchParameters = '?' + serializedRedirectSearchParameters
+        }
+        createEmptyBatchChange({
+            variables: { namespace: selectedNamespace.id, name: nameInput },
+        })
+            .then(args => {
+                if (!renderTemplate) {
+                    return Promise.resolve(args)
+                }
+
+                const template = renderTemplate(nameInput)
+
+                return args.data?.createEmptyBatchChange.id && template
+                    ? createBatchSpecFromRaw({
+                          variables: { namespace: selectedNamespace.id, spec: template, noCache: false },
+                      }).then(() => Promise.resolve(args))
+                    : Promise.resolve(args)
+            })
+            .then(({ data }) =>
+                data
+                    ? history.push(`${data.createEmptyBatchChange.url}/edit${serializedRedirectSearchParameters}`)
+                    : noop()
+            )
+            // We destructure and surface the error from `useMutation` instead.
+            .catch(noop)
+    }
+
+    return (
+        <Form className={styles.form} onSubmit={handleCreate}>
+            <Container className="mb-4">
+                {error && <ErrorAlert error={error} />}
+                <NamespaceSelector
+                    namespaces={namespaces}
+                    selectedNamespace={selectedNamespace.id}
+                    onSelect={setSelectedNamespace}
+                    disabled={isReadOnly}
+                />
+                <Input
+                    label="Batch change name"
+                    value={nameInput}
+                    onChange={onNameChange}
+                    pattern={String(NAME_PATTERN)}
+                    required={true}
+                    status={isNameValid === undefined ? undefined : isNameValid ? 'valid' : 'error'}
+                    placeholder="My batch change name"
+                    disabled={isReadOnly}
+                />
+                {!isReadOnly && (
+                    <small className="text-muted">
+                        Give it a short, descriptive name to reference the batch change on Sourcegraph. Do not include
+                        confidential information.{' '}
+                        <span className={classNames(isNameValid === false && 'text-danger')}>
+                            Only regular characters, _ and - are allowed.
```

Comment:
Sorry, I know you didn't add this in this PR, but:

```suggestion
                            Only letters, numbers, _ and - are allowed.
```